### ISSUE TITLE
Unable to send message

### ISSUE THREAD

**Issue by mohsinhussain** [RocketChat/Rocket.Chat.Android#39]
Just after your latest commit;
4fd15e938a8848d94267a047115437b8376b29a3
I am able to retrieve rooms data and its populating pretty well on side panel now. Chat history is also loading alright in chat window but when i send a new message to any room; private/public channel or direct message then my message is not sent and I get the following error in the debug log;

```
{"msg":"method","method":"sendMessage","params":[{"_id":"message-doc147756814805418","rid":"JPGi9me9T9ojQJviA","msg":"hello"}],"id":"message147756814805419"}

{"msg":"updated","methods":["message147756814805419"]}

{"msg":"result","id":"message147756814805419","error":{"error":"error-invalid-room","reason":"Invalid room","details":{"method":"canAccessRoom"},"message":"Invalid room [error-invalid-room]","errorType":"Meteor.Error"}}
```

`
Can you please explain why am I getting this error and why my messages are not going?

I am connecting to my custom server but this issue remains even on demo.rocket.chat. Any other client works fine on my custom server though.

**Comment by zeeshanqamar** [comment]
having the same issue, none of my messages are being sent

**Comment by zeeshanqamar** [comment]
chat.rocket.android E/Rocket.Chat.Android: error
                                                                          java.lang.UnsupportedOperationException
                                                                              at java.util.Collections$UnmodifiableCollection.add(Collections.java:928)
                                                                              at chat.rocket.android.fragment.MessageAdapter.insertAttachment(MessageAdapter.java:433)
                                                                              at chat.rocket.android.fragment.MessageAdapter.bindView(MessageAdapter.java:268)
                                                                              at chat.rocket.android.fragment.MessageAdapter.bindView(MessageAdapter.java:47)
                                                                              at chat.rocket.android.view.CursorRecyclerViewAdapter.onBindViewHolder(CursorRecyclerViewAdapter.java:59)
                                                                              at chat.rocket.android.fragment.MessageAdapter.onBindViewHolder(MessageAdapter.java:138)
                                                                              at chat.rocket.android.fragment.MessageAdapter.onBindViewHolder(MessageAdapter.java:47)
                                                                              at android.support.v7.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:6062)
                                                                              at android.support.v7.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:6095)
                                                                              at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5277)
                                                                              at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5153)
                                                                              at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2061)
                                                                              at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1445)
                                                                              at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1408)
                                                                              at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:555)
                                                                              at android.support.v7.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3374)
                                                                              at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:3183)
                                                                              at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:3627)
                                                                              at android.view.View.layout(View.java:15749)
                                                                              at android.view.ViewGroup.layout(ViewGroup.java:4880)
                                                                              at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1677)
                                                                              at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1531)
                                                                              at android.widget.LinearLayout.onLayout(LinearLayout.java:1440)
                                                                              at android.view.View.layout(View.java:15749)
                                                                              at android.view.ViewGroup.layout(ViewGroup.java:4880)
                                                                              at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
                                                                              at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
                                                                              at android.view.View.layout(View.java:15749)
                                                                              at android.view.ViewGroup.layout(ViewGroup.java:4880)
                                                                              at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
                                                                              at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
                                                                              at android.view.View.layout(View.java:15749)
                                                                              at android.view.ViewGroup.layout(ViewGroup.java:4880)
                                                                              at android.support.v4.widget.SlidingPaneLayout.onLayout(SlidingPaneLayout.java:725)
                                                                              at android.view.View.layout(View.java:15749)
                                                                              at android.view.ViewGroup.layout(ViewGroup.java:4880)
                                                                              at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
                                                                              at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
                                                                              at android.view.View.layout(View.java:15749)
                                                                              at android.view.ViewGroup.layout(ViewGroup.java:4880)
                                                                              at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
                                                                              at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
                                                                              at android.view.View.layout(View.java:15749)
                                                                              at android.view.ViewGroup.layout(ViewGroup.java:4880)
                                                                              at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
                                                                              at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
                                                                              at android.view.View.layout(View.java:15749)
                                                                              at android.view.ViewGroup.layout(ViewGroup.java:4880)
                                                                              at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1677)
                                                                              at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1531)
                                                                              at android.widget.LinearLayout.onLayout(LinearLayout.java:1440)
                                                                              at android.view.View.layout(View.java:15749)
                                                                              at android.view.ViewGroup.layout(ViewGroup.java:4880)
                                                                              at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
                                                                              at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
                                                                              at android.view.View.layout(View.java:15749)
                                                                              at android.view.ViewGroup.layout(ViewGroup.java:4880)
                                                                              at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2358)
                                                                            at android.view.ViewRootImpl.performTraversals(

**Comment by YusukeIwaki** [comment]
It will be re-implemented in #55.
